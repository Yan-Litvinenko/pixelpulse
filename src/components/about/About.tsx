import React from 'react';
import humanAvif from '../../assets/images/human.avif';
import humanJpg from '../../assets/images/human.jpg';
import humanWebp from '../../assets/images/human.webp';
import styles from './About.module.scss';
import { AboutElement } from '../aboutElement/AboutElement';
import { Frame } from '../frame/Frame';
import { useSelector } from 'react-redux';
import type { RootState } from '../../store/store';

const About = (): React.JSX.Element => {
    const { isSmall } = useSelector((state: RootState) => state.mediaQuery);

    return (
        <section className={styles.about}>
            <h2 className={styles.about__title}>who is Yan Litvinenko</h2>

            <div className={styles.about__content}>
                <div className={styles.about__text_content}>
                    <AboutElement
                        textContent={
                            'for the last half of the year i have been immersed in the world of frontend development. my journey started with javascript. soon i started learning react and typescript. i have always been interested in creating a user-friendly and interactive user interface.'
                        }
                        title={'The short introduction of my life'}
                    />
                    <AboutElement
                        textContent={
                            'although i like front end development, i am looking to depend my knowledge in backend development, especially with node.js and databases such as mysql and mongodb. i believe understanding the full stack will make me a more versatile and effective developer.'
                        }
                        title={'Career and development'}
                    />
                    <AboutElement
                        textContent={
                            'My goal is to become a well-rounded full-stack developer. I want to build applications that are not only visually appealing and user-friendly but also robust and efficient on the server side. I am committed to continuous learning and excited to see where this journey takes me.'
                        }
                        title={'More can be added in the left side for summary'}
                    />
                </div>

                {isSmall ? null : (
                    <picture
                        className={styles.box_image}
                        data-testid="about-mobile-picture"
                    >
                        <Frame className={styles.frame} />

                        <source srcSet={humanAvif} />
                        <source srcSet={humanWebp} />
                        <img
                            src={humanJpg}
                            alt="A portrait of Yan Litvinenko"
                            className={styles.about__img}
                        />
                    </picture>
                )}
            </div>
        </section>
    );
};

export { About };
