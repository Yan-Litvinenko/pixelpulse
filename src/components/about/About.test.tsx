import React from 'react';
import { About } from './About';
import { render, screen } from '@testing-library/react';
import type { RenderResult } from '@testing-library/react';
import * as reduxHooks from 'react-redux';

jest.mock('react-redux');

const mockUseSelector = jest.spyOn(reduxHooks, 'useSelector');

describe('About component', (): void => {
    test('Desktop about page', (): void => {
        mockUseSelector.mockReturnValue({ isSmall: false });

        const aboutComponent: RenderResult = render(<About />);

        expect(screen.getByTestId('about-mobile-picture')).toBeInTheDocument();
        expect(aboutComponent).toMatchSnapshot();
    });

    test('Mobile about page', (): void => {
        mockUseSelector.mockReturnValue({ isSmall: true });

        const aboutComponent: RenderResult = render(<About />);

        expect(
            screen.queryByTestId('about-mobile-picture'),
        ).not.toBeInTheDocument();
        expect(aboutComponent).toMatchSnapshot();
    });
});
