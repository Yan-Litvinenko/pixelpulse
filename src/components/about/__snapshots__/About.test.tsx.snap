// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component Desktop about page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <h2>
          who is Yan Litvinenko
        </h2>
        <div>
          <div>
            <article>
              <h3>
                The short introduction of my life
              </h3>
              <p>
                for the last half of the year i have been immersed in the world of frontend development. my journey started with javascript. soon i started learning react and typescript. i have always been interested in creating a user-friendly and interactive user interface.
              </p>
            </article>
            <article>
              <h3>
                Career and development
              </h3>
              <p>
                although i like front end development, i am looking to depend my knowledge in backend development, especially with node.js and databases such as mysql and mongodb. i believe understanding the full stack will make me a more versatile and effective developer.
              </p>
            </article>
            <article>
              <h3>
                More can be added in the left side for summary
              </h3>
              <p>
                My goal is to become a well-rounded full-stack developer. I want to build applications that are not only visually appealing and user-friendly but also robust and efficient on the server side. I am committed to continuous learning and excited to see where this journey takes me.
              </p>
            </article>
          </div>
          <picture
            data-testid="about-mobile-picture"
          >
            <div>
              <svg
                fill="none"
                height="0.5625rem"
                viewBox="0 0 9 9"
                width="0.5625rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1 9L1 1L9 1"
                  stroke="current-color"
                  stroke-width="2"
                />
              </svg>
              <svg
                fill="none"
                height="0.5625rem"
                viewBox="0 0 9 9"
                width="0.5625rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 1L8 1L8 9"
                  stroke="current-color"
                  stroke-width="2"
                />
              </svg>
              <svg
                fill="none"
                height="0.5625rem"
                viewBox="0 0 9 9"
                width="0.5625rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 0L8 8L1.04907e-06 8"
                  stroke="current-color"
                  stroke-width="2"
                />
              </svg>
              <svg
                fill="none"
                height="0.5625rem"
                viewBox="0 0 9 9"
                width="0.5625rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 8L1 8L0.999999 6.99382e-07"
                  stroke="current-color"
                  stroke-width="2"
                />
              </svg>
            </div>
            <source
              srcset="test-file-stub"
            />
            <source
              srcset="test-file-stub"
            />
            <img
              alt="A portrait of Yan Litvinenko"
              src="test-file-stub"
            />
          </picture>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <h2>
        who is Yan Litvinenko
      </h2>
      <div>
        <div>
          <article>
            <h3>
              The short introduction of my life
            </h3>
            <p>
              for the last half of the year i have been immersed in the world of frontend development. my journey started with javascript. soon i started learning react and typescript. i have always been interested in creating a user-friendly and interactive user interface.
            </p>
          </article>
          <article>
            <h3>
              Career and development
            </h3>
            <p>
              although i like front end development, i am looking to depend my knowledge in backend development, especially with node.js and databases such as mysql and mongodb. i believe understanding the full stack will make me a more versatile and effective developer.
            </p>
          </article>
          <article>
            <h3>
              More can be added in the left side for summary
            </h3>
            <p>
              My goal is to become a well-rounded full-stack developer. I want to build applications that are not only visually appealing and user-friendly but also robust and efficient on the server side. I am committed to continuous learning and excited to see where this journey takes me.
            </p>
          </article>
        </div>
        <picture
          data-testid="about-mobile-picture"
        >
          <div>
            <svg
              fill="none"
              height="0.5625rem"
              viewBox="0 0 9 9"
              width="0.5625rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 9L1 1L9 1"
                stroke="current-color"
                stroke-width="2"
              />
            </svg>
            <svg
              fill="none"
              height="0.5625rem"
              viewBox="0 0 9 9"
              width="0.5625rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 1L8 1L8 9"
                stroke="current-color"
                stroke-width="2"
              />
            </svg>
            <svg
              fill="none"
              height="0.5625rem"
              viewBox="0 0 9 9"
              width="0.5625rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0L8 8L1.04907e-06 8"
                stroke="current-color"
                stroke-width="2"
              />
            </svg>
            <svg
              fill="none"
              height="0.5625rem"
              viewBox="0 0 9 9"
              width="0.5625rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 8L1 8L0.999999 6.99382e-07"
                stroke="current-color"
                stroke-width="2"
              />
            </svg>
          </div>
          <source
            srcset="test-file-stub"
          />
          <source
            srcset="test-file-stub"
          />
          <img
            alt="A portrait of Yan Litvinenko"
            src="test-file-stub"
          />
        </picture>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`About component Mobile about page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <h2>
          who is Yan Litvinenko
        </h2>
        <div>
          <div>
            <article>
              <h3>
                The short introduction of my life
              </h3>
              <p>
                for the last half of the year i have been immersed in the world of frontend development. my journey started with javascript. soon i started learning react and typescript. i have always been interested in creating a user-friendly and interactive user interface.
              </p>
            </article>
            <article>
              <h3>
                Career and development
              </h3>
              <p>
                although i like front end development, i am looking to depend my knowledge in backend development, especially with node.js and databases such as mysql and mongodb. i believe understanding the full stack will make me a more versatile and effective developer.
              </p>
            </article>
            <article>
              <h3>
                More can be added in the left side for summary
              </h3>
              <p>
                My goal is to become a well-rounded full-stack developer. I want to build applications that are not only visually appealing and user-friendly but also robust and efficient on the server side. I am committed to continuous learning and excited to see where this journey takes me.
              </p>
            </article>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <h2>
        who is Yan Litvinenko
      </h2>
      <div>
        <div>
          <article>
            <h3>
              The short introduction of my life
            </h3>
            <p>
              for the last half of the year i have been immersed in the world of frontend development. my journey started with javascript. soon i started learning react and typescript. i have always been interested in creating a user-friendly and interactive user interface.
            </p>
          </article>
          <article>
            <h3>
              Career and development
            </h3>
            <p>
              although i like front end development, i am looking to depend my knowledge in backend development, especially with node.js and databases such as mysql and mongodb. i believe understanding the full stack will make me a more versatile and effective developer.
            </p>
          </article>
          <article>
            <h3>
              More can be added in the left side for summary
            </h3>
            <p>
              My goal is to become a well-rounded full-stack developer. I want to build applications that are not only visually appealing and user-friendly but also robust and efficient on the server side. I am committed to continuous learning and excited to see where this journey takes me.
            </p>
          </article>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
